javascript/javascript3/week3/canvas.js

```diff
@@ -0,0 +1,66 @@
+        console.log('Canvas dawing in JS');
+const c = document.getElementById("myCanvas");
+const ctx0 = c.getContext("2d");
+ctx0.beginPath();
+ctx0.arc(95,50,40,0,2*Math.PI);
+ctx0.stroke();
+ctx0.fillStyle = "blue";
+ctx0.fill();
+
+//  class creation time
+const ctx = c.getContext("2d");
+ctx.canvas.width  = window.innerWidth; // canvas size as width of screen
+ctx.canvas.height = window.innerHeight;// canvas size as height of screen
+class Circle{
+    constructor(x, y, r, startAngle, endAngle, fillColor){
+    this.x=x;
+    this.y=y;
+    this .radius=r;
+    this.stAngle=startAngle;
+    this.endAngle=endAngle;
+    this.color=fillColor;
+    
+}
+
+draw(){
+
+ctx.beginPath();
+ctx.arc(this.x,this.y,this .radius,this.stAngle, this.endAngle);
+ctx.stroke();
+ctx.fillStyle = this.color;
+ctx.fill();
+
+}
+    
+}
+const c1 = new Circle(50, 50, 20, 0, 2 * Math.PI, "#000000");
+c1.draw();
+const c2 = new Circle(150, 150, 80, 0, 2 * Math.PI, "#000000");
+c2.draw();
+const c3 = new Circle(350, 350, 120, 0, 2 * Math.PI, "#000000");
+c3.draw();
+
+
+// Now lets make art!=> Random circles creation in Canvas
+
+setInterval(()=>{
+    const x=Math.floor(Math.random() * 2000) + 1; 
+    const y=Math.floor(Math.random() * 2000) + 1; 
+    const r=Math.floor(Math.random() * 30) ; 
+    const randomColor = '#'+Math.floor(Math.random()*16777215).toString(16);
+    const c1 = new Circle(x, y, r, 0, 2 * Math.PI, randomColor);
+    c1.draw();
+},100);
+
+// drawing circles on mouse move
+document.getElementById("myCanvas").addEventListener("mousemove", function(event) {
+    setInterval(()=>{
+        const x=event.clientX; 
+        const y=event.clientY;
+        const r=Math.floor(Math.random() * 30) ; 
+        const randomColor = '#'+Math.floor(Math.random()*16777215).toString(16);
+        const c1 = new Circle(x, y, r, 0, 2 * Math.PI, randomColor);
+        c1.draw();
+    },1000);
+  });
```

Comment:
Great job and beautiful art! I tried to run your code and it works. I have one suggestion: to make some changes in the part where you are drawing the circles following the mouse move. Here I tried to do the same thing to insert setInterval inside the addEventListener, and as far as I understand it is causing some problems. It listens to your move and then replicates this move every next second, even when the mouse is not moving anymore. Sorry for my poor explanation, but my point is - to remove setInterval from the inside of addEventListener, and place it separately. In this case, it will draw circles following your moves on the screen, and it will stop when you stop moving the mouse.

Reply:
thanks for the solution :) It is fixed now :)